### PR TITLE
clean up workload attest metrics

### PR DESCRIPTION
**Pull request by amoore877**
Signed-off-by: Andrew Moore <andrew.s.moore@uber.com>

**Pull Request check list**

- [X] Commit conforms to CONTRIBUTING.md?
- [X] Proper tests/regressions included?
- [X] Documentation updated?

**Affected functionality**
Workload Attest metrics

**Description of change**
`workload_attestor_latency` measures individual attestor latency, and is currently used for `CallCounter`s, which means we emit a `workload_attestor_latency` metric which is actually a call count, and a `workload_attestor_latency_elapsed_time` metric which is the actual latency. Remove `latency` from the metric name.

`workload_attest_duration` is a simple latency measure for the entirety of attestation, but is too close of a name to `workload_attestor_latency`; change this to be a `CallCounter` and drop the `duration` name.

end result:
- individual attestors have `workload_attestor` count and `workload_attestor_elapsed_time` latency, both with an attestor type tag
- overall attestation has `workload_attestation` count and `workload_attestation_elapsed_time` latency

**Which issue this PR fixes**
n/a

### CODE REVIEW

File: pkg/agent/attestor/workload/workload.go [hunk 1]
```diff
@@ -38,7 +37,6 @@ type Config struct {
 // Attest invokes all workload attestor plugins against the provided PID. If an error
 // is encountered, it is logged and selectors from the failing plugin are discarded.
 func (wla *attestor) Attest(ctx context.Context, pid int32) []*common.Selector {
-	defer telemetry_workload.MeasureAttestDuration(wla.c.Metrics, time.Now())
```
Comment (evan2645) [inline pkg/agent/attestor/workload/workload.go]: I think the difference between this and the below invocation of `StartAttestorLatencyCall` is that the latter measures attestation latency on a per-plugin basis, and this defer measures the duration of attestation as a whole.

Comment (amoore877) [reply pkg/agent/attestor/workload/workload.go]: the only thing the deleted metric would be measuring is the beginning work of getting the attestors and aggregating of the selectors, neither of which I think would be computationally very interesting compared to the actual attestations

we *could* still keep this, but we definitely need better naming. `Workload Attestor Latency` and `Workload Attestation Duration` are too close of names, and for the former does not take advantage of the existence of the CallCounter `Done` logic adding an `elapsed_time` tag

if we want to keep, bare minimum I think would be dropping `latency` and `duration` from the names, and turning the above overall measurement into a CallCounter to get the full latency + call count view. Even better would be if we can think of namings that are not so close to each other

Comment (azdagron) [reply pkg/agent/attestor/workload/workload.go]: Without this metric, it seems hard to figure out the total time taken for attestation without doing some fancy summing... knowing the total attestation time seems like the primary metric I'd be interested in. The per-plugin metrics would then be helpful in drilling down. My vote is to keep it.

Comment (amoore877) [reply pkg/agent/attestor/workload/workload.go]: That's a good point, I would argue still on dropping `latency` from the one name and (given your statement on benefits of aggregation) changing the overall metric to use a call counter and dropping `duration` from the name.
`workload_attestor` with a specific `attestor type` flag vs. `workload_attestation` may be good enough distinction.
what do you think?

Comment (azdagron) [reply pkg/agent/attestor/workload/workload.go]: I like it.

**Pull request merged**